tests/utils_tests.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file dataloader_tests.cpp
+ * @author Kartik Dutt
+ *
+ * Tests for various functionalities of utils.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BOOST_TEST_DYN_LINK
+#include <utils/utils.hpp>
+#include <dataloader/datasets.hpp>
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(UtilsTest);
+
+/**
+ * Simple test for Data Downloader.
+ */
+BOOST_AUTO_TEST_CASE(DownloadFileTest)
+{
+  Utils::DownloadFile(Datasets::MNIST().trainDownloadUrl, "./../data/mnist_train.csv", "", false);
+  BOOST_REQUIRE(Utils::PathExists("./../data/mnist_train.csv"));
+  BOOST_REQUIRE(Utils::CompareSHA256("./../data/mninst_train.csv", Datasets::MNIST().trainHash));
+
+  Utils::DownloadFile(Datasets::MNIST().testDownloadUrl, "./../data/mnist_test.csv", "", false);
+  BOOST_REQUIRE(Utils::PathExists("./../data/mnist_test.csv"));
+  BOOST_REQUIRE(Utils::CompareSHA256("./../data/mninst_test.csv", Datasets::MNIST().testHash));
```

Comment:
Should we remove the files we downloaded with `remove()` after running the tests?  (Also should this line compare with `mnist_test.csv` not `mninst_test.csv`?)

Reply:
Sounds like a good idea. Will add it to the implementation.